lib/instrumentation/http-shared.js

```diff
@@ -152,23 +153,25 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
       // If so, then fallback to the current run context's span or transaction,
       // if any.
       var parent = span || parentRunContext.currSpan() || parentRunContext.currTransaction()
-      if (parent && parent._context) {
-        const headerValue = parent._context.toTraceParentString()
-        const traceStateValue = parent._context.toTraceStateString()
-
-        const newHeaders = Object.assign({}, options.headers)
-        newHeaders.traceparent = headerValue
-        if (traceStateValue) {
-          // Do not set the header to the empty string. Though RFC 7230 allows
-          // empty header values, some servers apparently do not.
-          newHeaders.tracestate = traceStateValue
+
+      const setter = function (carrier, name, value) {
+        if (!value) {
+          return
         }
-        if (agent._conf.useElasticTraceparentHeader) {
-          newHeaders['elastic-apm-traceparent'] = headerValue
+        carrier[name] = value
+
+        if (agent._conf.useElasticTraceparentHeader && name === 'traceparent') {
+          carrier['elastic-apm-traceparent'] = value
         }
-        options.headers = newHeaders
       }
-
+      const newHeaders = Object.assign({}, options.headers)
+      if (parent && parent._context && span) {
+        parent.propagateTraceContextHeaders(newHeaders, setter)
+      } else if (parent && parent._context && !span) {
+        // headers still need to propagate
+        propagateTraceContextHeaders(newHeaders, parent._context, setter)
+      }
```

Comment:
Isn't the else branch redundant? The `span` variable also gets assigned to the `parent` variable, if present.
```suggestion
      if (parent) {
        parent.propagateTraceContextHeaders(newHeaders, setter)
      }
```

Reply:
cbd1cd80fe46cc43c0fd89828a20e72874cc4d08

Reply:
Hmm -- I guess with the current semantics that branch is redundant and the tests seem happy with the change.  Getting rid of it also means we can avoid exporting that logic as a stand alone function.  

Also, not explicitly mentioned, but re: the `_context` check you removed we've been cargo culting that `parent._context` check along for a while now, but it seems reasonable to remove it (as along as `currSpan` or `currTransaction` keep returning something that has GenericSpan as a base object)

Done in cbd1cd80fe46cc43c0fd89828a20e72874cc4d08